### ISSUE TITLE
Add packages without changing original source

### ISSUE THREAD

**Issue by matslil** [cpp-pm/hunter#255]
* I've read [Brief overview section](https://hunter.readthedocs.io/en/latest/overview.html) and do understand basic concepts. **Yes**
* I've read [F.A.Q. section](https://hunter.readthedocs.io/en/latest/faq.html) and haven't found an answer to my question. **Yes**
* I've read [Code of Conduct](https://github.com/cpp-pm/hunter/blob/master/.github/CODE_OF_CONDUCT.md), I promise to be polite and will do my best at being constructive. **Yes**

---

Would it be possible to use CMAKE_TOOLCHAIN_FILE as a way to "patch" CMake projects? Apart from the usual work that a toolchain file needs to do, when adding external projects it could also:

- Make sure HunterGate is initialised properly
- Overload find_package(), so hunter_add_package() is called when appropriate

This would not solve all problems that forks solve, but might mean that fewer projects need to be forked.

Would this work?

**Comment by rbsheth** [comment]
I don't think we can use the toolchain file for this. However, we can have instructions to patch sources after download to avoid the creation of forks. Anything to make the process of adding packages here simpler is welcomed!
